pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -306,7 +306,7 @@ func getVMOverrideForIstiodDNS(ctx resource.Context, cfg echo.Config) (istioHost
 	istioRevision := getIstioRevision(cfg.Namespace)
 	istioHost = istioctlcmd.IstiodHost(istioNS, istioRevision)
 
-	istioIPAddr := ist.EastWestGatewayFor(cfg.Cluster).DiscoveryAddress().Addr()
+	istioIPAddr := ist.EastWestGatewayFor(cfg.Cluster).DiscoveryAddresses()[0].Addr()
```

Comment:
I tried to update the framework to use DiscoveryAddresses everywhere but I realized that it would end up leading to more extensive changes to how istiod is deployed on VMs.
I have for now updated the other scenarios while keeping DiscoveryAddresses-related ones untouched